src/mixins/add-edit-org-person-mixin.ts

```diff
@@ -312,15 +312,15 @@ export default class AddEditOrgPersonMixin extends Vue {
   protected async updateBusinessDetails (businessLookup: BusinessLookupIF): Promise<void> {
     this.orgPerson.officer.organizationName = businessLookup.name
     this.orgPerson.officer.identifier = businessLookup.identifier
-    businessLookup.bn = businessLookup.bn?.length > 9
-      ? businessLookup.bn.slice(0, 9)
-      : businessLookup.bn
-    this.orgPerson.officer.businessNumber = businessLookup.bn
-    this.orgPerson.showOptionalBN = !businessLookup.bn
+    // sanitize Business Number
+    this.orgPerson.officer.businessNumber = (businessLookup.bn?.length === 9) ? businessLookup.bn
+      : (businessLookup.bn?.length > 9) ? businessLookup.bn.slice(0, 9) : null
+    this.orgPerson.showOptionalBN = !this.orgPerson.officer.businessNumber
 
     this.inProgressBusinessLookup = { ...businessLookup }
     if (businessLookup.identifier) {
       const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
+        .catch(() => ({ registeredOffice: undefined }))
```

Comment:
Catching the error changes this:

![image](https://user-images.githubusercontent.com/10002889/177433748-f9765a32-0d7c-4ce7-a834-6a4eac4894fa.png)

to this:

![image](https://user-images.githubusercontent.com/10002889/177433831-d38aa4b2-e08b-4a47-bfa2-b1dd8df6d825.png)